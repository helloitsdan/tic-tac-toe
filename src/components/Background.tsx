import React, { FunctionComponent, useMemo } from 'react'
import { useSelector } from 'react-redux'
import classnames from 'classnames'
import { getWinner } from '../store/selectors'
import { PlayerId } from '../types'

const Background: FunctionComponent = () => {
  const winner = useSelector(getWinner)

  const winnerId = useMemo(() => {
    if (!winner) {
      return null
    }

    return winner.player
  }, [winner])

  return (
    <div
      className={classnames('c-background', {
        'c-background--player-one': winnerId === PlayerId.PLAYER_ONE,
        'c-background--player-two': winnerId === PlayerId.PLAYER_TWO,
      })}
    />
  )
}

export default Background
