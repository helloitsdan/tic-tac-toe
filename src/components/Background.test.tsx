import React from 'react'
import { render } from '@testing-library/react'

import Background from './Background'
import { createStore } from 'redux'
import { Provider } from 'react-redux'

describe('Background', () => {
  it('renders', () => {
    const testStore = createStore(() => ({}))

    const { asFragment } = render(
      <Provider store={testStore}>
        <Background />
      </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
