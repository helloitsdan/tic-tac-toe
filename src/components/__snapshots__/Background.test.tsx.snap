// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background renders 1`] = `
<DocumentFragment>
  <div
    class="c-background"
  />
</DocumentFragment>
`;
